### PR TITLE
aro install reherse job

### PR DESCRIPTION
**Pull request by mgahagan73**
rehersal run for ARO installer, please do not merge, Work in proress

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-commands.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+echo "Running ARO create command:"
+echo "${CREATE_CMD}"
+
+AROINFO="$(eval "$CREATE_CMD")"
+
+echo "Cluster created, sleeping 600";
+
+sleep 600
+
+echo "$AROINFO" > ${SHARED_DIR}/clusterinfo
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-commands.sh:60]: Seem like we can directly `eval "$CREATE_CMD" > ${SHARED_DIR}/clusterinfo` to skip `AROINFO`

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+	"path": "cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json:2]: 1. Is it possible aro vnet re-use the `azure-provision-vnet` step as common OCP install?
2. if no, probably we need to place the step in `ci-operator/step-registry` folder (e.g: `ci-operator/step-registry/azure/provision/aro-net`), `cucushift/installer/rehearse` folder is for saving cluster install workflow yaml files.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-ref.metadata.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+	"path": "cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-ref.metadata.json:2]: cucushift/installer/rehearse folder is for saving cluster install workflow yaml files, probably we need to place the step in ci-operator/step-registry folder (e.g: ci-operator/step-registry/azure/provision/aro).

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json (hunk 2 above)
Comment (mgahagan73) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json:2]: The aro vnet required is slightly different than the standard Azure one, it probably makes sense to give it its own step.

We have ci-operator/step-registry/aro/  which got checked in by mistake in a previous PR, I will see if I can move all the steps there and get it working again.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-commands.sh (hunk 1 above)
Comment (mgahagan73) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-commands.sh:60]: fixed

**Comment by mgahagan73** [comment]
still looking at this one, tests reported as passing but with the time it took there is no way it could have provisioned/removed an ARO cluster

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-ref.metadata.json (hunk 3 above)
Comment (mgahagan73) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/provision/cucushift-installer-rehearse-azure-aro-provision-ref.metadata.json]: I've moved the aro vnet and cluster provision code to step-registry/aro/provision

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json (hunk 2 above)
Comment (mgahagan73) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/vnet/cucushift-installer-rehearse-azure-aro-vnet-ref.metadata.json]: vnet provioner now lives in ci-operator/step-registry/aro/provision/vnet

**Pull request merged**